apiserver/facades/client/application/deployrepository_test.go

```diff
@@ -650,7 +651,11 @@ func (s *deployRepositorySuite) TestDeployFromRepositoryAPI(c *gc.C) {
 		Charm: template.charm,
 		ID:    charm.MustParseURL("ch:amd64/jammy/testme-5"),
 	}
-	s.state.EXPECT().AddCharmMetadata(info).Return(&state.Charm{}, nil)
+
+	s.charm.EXPECT().Meta().Return(&charm.Meta{Resources: nil})
+
+	s.state.EXPECT().AddCharmMetadata(info).Return(s.charm, nil)
+
 	addAppArgs := state.AddApplicationArgs{
 		Name:  "metadata-name",
 		Charm: &state.Charm{},
```

Comment:
Add a comment about why we're putting a state.charm pointer to an empty struct here.